docs/next/components/SidebarNavigation.tsx

```diff
@@ -0,0 +1,291 @@
+import cx from 'classnames';
+import {useEffect, useState} from 'react';
+import GitHubButton from 'react-github-btn';
+import visit from 'unist-util-visit';
+
+// ==============================
+// handling Markdoc-style Markdown
+// ==============================
+
+export function collectHeadings(node, headings = []) {
```

Comment:
Can this file be TS-annotated throughout? It looks like it might be a lot of code copied over from elsewhere, but I think it'd be nice to lock down the types if we can.